torch_xla/csrc/tensor.cpp

```diff
@@ -291,16 +299,6 @@ void XLATensor::SetInPlaceIrValue(torch::lazy::Value ir_value) {
 }
 
 void XLATensor::AssignIrValue(torch::lazy::Value ir_value) const {
-  ShardingSpecPtr sharding = sharding_spec();
-  if (sharding != nullptr) {
-    if (!ir_value) {
-      // Create a tensor node if applicable, re-use the current IR otherwise.
-      // TODO(yeounoh) this has some performance implications for convolution.
-      ir_value = GetIrValue();
-    }
-    dynamic_cast<XlaNode*>(ir_value.node.get())
-        ->SetSharding(sharding->sharding);
-  }
   data()->ir_value = std::move(ir_value);
   data()->generation += 1;
```

Comment:
what if passed in `ir_value` has a sharding spec?

Reply:
The check will be done at the sharding_spec() call.

Reply:
I am afraid this will make debugging a bit difficult, since we can only see that there is a mismatch but it is hard to figure out why(error message is lazy). The code is cleaner this way I guess, so I am also OK with this approach.

Reply:
Yea, it's cleaner this way that we have a limited (consolicated) places, where we do the checks and the syncing. I see your point, though. Let's add some debugging probes `TF_VLOG` for that instead.

Reply:
let's do
```
  TF_VLOG(5) << "Assign IR value " << ir_value->ToString() + sharding ? (" with sharding " + sharding->DebugString()) : "";
```
This way when `TF_VLOG` is not enabled, we don't construct the debug string.